parser.y

```diff
@@ -8251,47 +8259,134 @@ SelectStmtLimitOpt:
 	}
 |	SelectStmtLimit
 
+SelectStmtOpt:
+	TableOptimizerHints
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		opt.TableHints = $1.([]*ast.TableOptimizerHint)
+		$$ = opt
+	}
+|	DistinctOpt
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		if $1.(bool) {
+			opt.Distinct = true
+		} else {
+			opt.Distinct = false
+			opt.ExplicitAll = true
+		}
+		$$ = opt
+	}
+|	Priority
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		opt.Priority = $1.(mysql.PriorityEnum)
+		$$ = opt
+	}
+|	SelectStmtSQLSmallResult
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		opt.SQLSmallResult = $1.(bool)
+		$$ = opt
+	}
```

Comment:
just use the keyword directly? it's not like the non-terminal can produce anything other than `true`.

```suggestion
|	"SQL_SMALL_RESULT"
	{
		opt := &ast.SelectStmtOpts{}
		opt.SQLCache = true
		opt.SQLSmallResult = true
		$$ = opt
	}
```

(ditto for the rest of this list)